File: Ser2023/Company/Company.Web/Modules/Company/PersonView/PersonViewForm.cs
using Serenity.ComponentModel;

namespace Company.Company.Forms;

[FormScript("Company.PersonView")]
[BasedOnRow(typeof(PersonViewRow), CheckNames = true)]
public class PersonViewForm
{
    public string PersonType { get; set; }
    public string Title { get; set; }
    public string FirstName { get; set; }
    public string MiddleName { get; set; }
    public string LastName { get; set; }
    public string Suffix { get; set; }
    public string Name { get; set; }
    public string PhoneNumberType { get; set; }
    public string AddressType { get; set; }
    public string AddressLine1 { get; set; }
    public string AddressLine2 { get; set; }
    public string City { get; set; }
    public string PostalCode { get; set; }
    public string StateProvinceCode { get; set; }
    public string Country { get; set; }
}

File: Ser2023/Company/Company.Web/Modules/ServerTypes/Company/PersonViewForm.ts
import { StringEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface PersonViewForm {
    PersonType: StringEditor;
    Title: StringEditor;
    FirstName: StringEditor;
    MiddleName: StringEditor;
    LastName: StringEditor;
    Suffix: StringEditor;
    Name: StringEditor;
    PhoneNumberType: StringEditor;
    AddressType: StringEditor;
    AddressLine1: StringEditor;
    AddressLine2: StringEditor;
    City: StringEditor;
    PostalCode: StringEditor;
    StateProvinceCode: StringEditor;
    Country: StringEditor;
}

export class PersonViewForm extends PrefixedContext {
    static formKey = 'Company.PersonView';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!PersonViewForm.init)  {
            PersonViewForm.init = true;

            var w0 = StringEditor;

            initFormType(PersonViewForm, [
                'PersonType', w0,
                'Title', w0,
                'FirstName', w0,
                'MiddleName', w0,
                'LastName', w0,
                'Suffix', w0,
                'Name', w0,
                'PhoneNumberType', w0,
                'AddressType', w0,
                'AddressLine1', w0,
                'AddressLine2', w0,
                'City', w0,
                'PostalCode', w0,
                'StateProvinceCode', w0,
                'Country', w0
            ]);
        }
    }
}